docs/DeveloperGuide.adoc

```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+3. User requests to filter expenses by specific tag
+4. Financial tracker shows a list of recent expenses under the specific tag.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no existing tags
++
+[none]
+** 2a1. Financial tracker displays an empty list
++
+Use case ends.
+
+* 3a. The given tag name is invalid
++
+[none]
+** 3a1.  Financial Tracker shows an error message.
++
+Use case resumes at step 2.
+
+
+[Discrete]
+=== UC03: Add an expense to an archive
+
+*MSS*
+
+1. User requests to list recent expenses
+2. Financial Tracker shows a list of recent expenses
+3. User requests to list all archives
+4. Financial Tracker shows a list of archives
+5. User requests to add a specific expense into a specific archive
+6. Financial Tracker removes the specific  expense from list of current records and adds it to the specific  archive
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There is no current records
++
+[none]
+** 2a1. Financial tracker displays an empty list
++
+Use case ends.
+
+* 4a. There are no existing archives
++
+[none]
+** 4a1.  Financial Tracker shows an empty list.
++
+Use case ends.
+
+* 5a. The given record index or archive name is invalid
++
+[none]
+** 5a1.  Financial Tracker shows an error message.
++
+Use case resumes at step 4.
+
+
+[Discrete]
+=== UC04: Deleting an archived expense
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to list all expenses under a specific archive
+2. Financial Tracker shows a list of expenses under the specific archive
+3. User requests to delete a specific expense from the archive
+4. Financial Tracker deletes the specific  expense from the specific  archive.
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 2a. The archive has no expenses in it
++
+[none]
+** 2a1. Financial tracker displays an empty list
++
+Use case ends.
+
+* 2b. The archive name is invalid
++
+[none]
+** 2b1.  Financial Tracker shows an error.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. The given record index or archive name is invalid
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1.  Financial Tracker shows an error message.
 +
 Use case resumes at step 2.
 
+
+[Discrete]
+=== UC05: Deleting an entire archive
+
+*MSS*
+
+1. User requests to list all existing archives
+2. Financial Tracker shows a list of all existing archives
+3. User requests to delete a specific archive from the list
+4. Financial Tracker deletes the specific archive
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no existing archives
++
+[none]
+** 2a1. Financial tracker displays an empty list
++
+Use case ends.
+
+* 3a.  The given archive name is invalid
++
+[none]
+** 3a1.  Financial Tracker shows an error.
++
+Use case resumes at step 2.
+
+
+[Discrete]
+=== UC06: Finding a record with a specified keyword
```

Comment:
Maybe could use precondition of the keyword being valid characters (eg. English)